src/main/java/org/opensearch/flowframework/workflow/ReindexStep.java

```diff
@@ -0,0 +1,175 @@
+    public static final String NAME = "reindex";
+    /** The refresh field for reindex */
+    private static final String REFRESH = "refresh";
+    /** The requests_per_second field for reindex */
+    private static final String REQUESTS_PER_SECOND = "requests_per_second";
+    /** The require_alias field for reindex */
+    private static final String REQUIRE_ALIAS = "require_alias";
+    /** The slices field for reindex */
+    private static final String SLICES = "slices";
+    /** The max_docs field for reindex */
+    private static final String MAX_DOCS = "max_docs";
+
+    /**
+     * Instantiate this class
+     *
+     * @param client Client to create an index
+     * @param flowFrameworkIndicesHandler FlowFrameworkIndicesHandler class to update system indices
+     */
+    public ReindexStep(Client client, FlowFrameworkIndicesHandler flowFrameworkIndicesHandler) {
+        this.client = client;
+        this.flowFrameworkIndicesHandler = flowFrameworkIndicesHandler;
+    }
+
+    @Override
+    public PlainActionFuture<WorkflowData> execute(
+        String currentNodeId,
+        WorkflowData currentNodeInputs,
+        Map<String, WorkflowData> outputs,
+        Map<String, String> previousNodeInputs,
+        Map<String, String> params
+    ) {
+
+        PlainActionFuture<WorkflowData> reIndexFuture = PlainActionFuture.newFuture();
+
+        Set<String> requiredKeys = Set.of(SOURCE_INDICES, DESTINATION_INDEX);
+
+        Set<String> optionalKeys = Set.of(REFRESH, REQUESTS_PER_SECOND, REQUIRE_ALIAS, SLICES, MAX_DOCS);
+
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs,
+                params
+            );
+
+            String sourceIndices = (String) inputs.get(SOURCE_INDICES);
+            String destinationIndex = (String) inputs.get(DESTINATION_INDEX);
+            Boolean refresh = inputs.containsKey(REFRESH) ? Booleans.parseBoolean(inputs.get(REFRESH).toString()) : null;
+            Float requestsPerSecond = inputs.containsKey(REQUESTS_PER_SECOND)
+                ? Float.parseFloat(inputs.get(REQUESTS_PER_SECOND).toString())
+                : Float.POSITIVE_INFINITY;
+            Boolean requireAlias = inputs.containsKey(REQUIRE_ALIAS) ? Booleans.parseBoolean(inputs.get(REQUIRE_ALIAS).toString()) : null;
+            Integer slices = (Integer) inputs.get(SLICES);
+            Integer maxDocs = (Integer) inputs.get(MAX_DOCS);
+
+            ReindexRequest reindexRequest = new ReindexRequest().setSourceIndices(Strings.splitStringByCommaToArray(sourceIndices))
+                .setDestIndex(destinationIndex);
+
+            if (refresh != null) {
+                reindexRequest.setRefresh(refresh);
+            }
+            if (requestsPerSecond != null) {
```

Comment:
Since you used a default `Float.POSITIVE_INFINITY` above, this null check is useless.  However I think the better fix here is to have the "else" part of the line 95-97 calculation be `null` rather than assuming the default we only know because of implementation details.